benchmarks/fuzzy_join_benchmark.py

```diff
@@ -0,0 +1,299 @@
+from autofj import AutoFJ
+from dirty_cat._fuzzy_join import FuzzyJoin
+
+
+def fetch_data(dataset_name):
+    """ Fetch datasets from https://github.com/chu-data-lab/AutomaticFuzzyJoin/tree/master/src/autofj/benchmark """
+    left = pd.read_csv(
+        f"https://raw.githubusercontent.com/chu-data-lab/AutomaticFuzzyJoin/master/src/autofj/benchmark/{dataset_name}/left.csv"
+    )
+    right = pd.read_csv(
+        f"https://raw.githubusercontent.com/chu-data-lab/AutomaticFuzzyJoin/master/src/autofj/benchmark/{dataset_name}/right.csv"
+    )
+    gt = pd.read_csv(
+        f"https://raw.githubusercontent.com/chu-data-lab/AutomaticFuzzyJoin/master/src/autofj/benchmark/{dataset_name}/gt.csv"
+    )
+    return left, right, gt
+
+
+def thefuzz_merge(df_1, df_2, key1, key2, threshold=90, limit=2, scorer=partial_ratio):
+    """
+    Merging using thefuzz
+
+    Parameters:
+        df_1: the left table to join
+        df_2: the right table to join
+        key1: key column of the left table
+        key2: key column of the right table
+        threshold: how close the matches should be to return a match, based on Levenshtein distance
+        limit: the amount of matches that will get returned, these are sorted high to low
+
+    Return:
+        dataframe with boths keys and matches
+    """
+    s = df_2[key2].tolist()
+
+    m = df_1[key1].apply(lambda x: process.extract(x, s, limit=limit, scorer=scorer))
+    df_1["matches"] = m
+
+    m2 = df_1["matches"].apply(
+        lambda x: ", ".join([i[0] for i in x if i[1] >= threshold])
+    )
+
+    df_1["matches"] = m2
+
+    return df_1[df_1["matches"] != ""]
+
+
+def test_autofj(left, right, gt, target):
+    """ Merging using AutomaticFuzzyJoin """
+    autofj = AutoFJ(precision_target=target, verbose=True)
+    LR_joins = autofj.join(left, right, id_column="id")
+
+    gt_joins = gt[["id_l", "id_r"]].values
+    LR_joins = LR_joins[["id_l", "id_r"]].values
+    p, r, f1 = evaluate(LR_joins, gt_joins)
+    print("Precision:", p, "Recall:", r, "F1:", f1)
+    return LR_joins, gt_joins
+
+
+def evaluate(pred_joins, gt_joins):
+    """ Evaluate the performance of fuzzy joins
+
+    Parameters
+    ----------
+    pred_joins: list
+        A list of tuple pairs (id_l, id_r) that are predicted to be matches
+
+    gt_joins:
+        The ground truth matches
+
+    Returns
+    -------
+    precision: float
+        Precision score
+
+    recall: float
+        Recall score
+
+    f1: float
+        F1 score
+    """
+    pred = {(le, ri) for le, ri in pred_joins}
+    gt = {(le, ri) for le, ri in gt_joins}
+
+    tp = pred.intersection(gt)
+
+    precision = len(tp) / len(pred)
+    recall = len(tp) / len(gt)
+    # print('Precision', precision, 'Recall', recall)
+    if precision > 0 or recall > 0:
+        f1 = 2 * precision * recall / (precision + recall)
+    else:
+        f1 = 0
+    return precision, recall, f1
+
+
+def FuzzyJoin_precision_recall(left, right, gt, left_col, right_col):
```

Comment:
Rename this (and others) to remove capitalization: fuzzy_join_precision_recall, as this is a function, and not a class